usage/usage.go

```diff
@@ -0,0 +1,136 @@
+// Package usage implements usage tracking for k6 in order to figure what is being used within a given execution
+package usage
+
+import (
+	"strings"
+	"sync"
+)
+
+// Usage is a way to collect usage data for within k6
+type Usage struct {
+	l *sync.Mutex
+	m map[string]any
+}
+
+// New returns a new empty Usage ready to be used
+func New() *Usage {
+	return &Usage{
```

Comment:
Out of curiosity, why not sync map?

Reply:
I think the answer is that I did try it and then decided to have 2 maps per type ... and hten merged it back.


Looking at the API it seems using it will be harder to implement the current API. 

I would expect it will be *faster*, but I am not certain how relevant it is for this to be as fast as posibble :shrug: and maybe it won't be if we are touching the same keys either way.